test/models/parser/twitter_profile_test.rb

```diff
@@ -71,62 +60,123 @@ def empty_doc
+      .to_return(status: 500, body: response_fixture_from_file('twitter-profile-response-error.json'))
 
-    data = {}
     sentry_call_count = 0
     arguments_checker = Proc.new do |e|
       sentry_call_count += 1
-      assert_equal Twitter::Error, e.class
     end
-
+    
     PenderSentry.stub(:notify, arguments_checker) do
-      data = Parser::TwitterProfile.new('https://www.twitter.com/fakeaccount').parse_data(nil)
+      assert_raises ProviderTwitter::ApiError do
+        Parser::TwitterProfile.new('https://twitter.com/fake_user').parse_data(empty_doc)
+     end
       assert_equal 1, sentry_call_count
-    end
-    assert_match /Twitter::Error/, data['error']['message']
-    assert_equal "fakeaccount", data['title']
-    assert_equal "https://twitter.com/fakeaccount", data['url']
-    assert_equal "fakeaccount", data['external_id']
-    assert_equal "@fakeaccount", data['username']
-    assert_equal "fakeaccount", data['author_name']
+    end      
   end
 
   test "assigns values to hash from the API response" do
-    Twitter::REST::Client.any_instance.stubs(:user).returns(fake_twitter_user)
-
-    data = Parser::TwitterProfile.new('https://www.twitter.com/fakeaccount').parse_data(empty_doc)
-
-    assert_equal 'fakeaccount', data['external_id']
-    assert_equal '@fakeaccount', data['username']
-    assert_match /TED is a nonprofit devoted to spreading ideas/, data['description']
-    assert_match 'TED Talks', data['title']
-    assert_match 'TED Talks', data['author_name']
-
-    assert_match 'https://twitter.com/fakeaccount', data['url']
-    assert_match /pbs.twimg.com\/profile_images\/877631054525472768\/Xp5FAPD5.jpg/, data['picture']
-    assert_match /pbs.twimg.com\/profile_images\/877631054525472768\/Xp5FAPD5.jpg/, data['author_picture']
+    stub_profile_lookup.returns(twitter_profile_response_success)
+    
+    data = Parser::TwitterProfile.new('https://www.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_equal 'fake_user', data['external_id']
+    assert_equal '@fake_user', data['username']
+    assert_match /The world's most powerful space telescope/, data['description']
+    assert_match 'fake_user', data['title']
+    assert_match 'Fake User', data['author_name']
+    assert_match 'https://twitter.com/fake_user', data['url']
+    assert_match /pbs.twimg.com\/profile_images\/685182791496134658\/Wmyak8D6.jpg/, data['picture']
+    assert_match /pbs.twimg.com\/profile_images\/685182791496134658\/Wmyak8D6.jpg/, data['author_picture']
     assert_not_nil data['published_at']
-
     assert_nil data['error']
+  end  
+  
+  test "should store raw data of profile returned by Twitter API" do
+    stub_profile_lookup.returns(twitter_profile_response_success)
+    
+    data = Parser::TwitterProfile.new('https://www.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_not_nil data['raw']['api']
+    assert !data['raw']['api'].empty?
+  end  
+  
+  test "returns data even if an error is returned" do
```

Comment:
**question** does twitter return 200 and then still return an errored json response? if so, we have a few error cases (writing this out mostly because I only just realized, so that lack of previous understanding may be present in my previous comments):
1. 200 response with error in json (which you've handled)
1. non-200 response
1. exception in Net::HTTP

this seems like a good test for the first case.